tv/tv.cpp

```diff
@@ -359,7 +381,9 @@ struct TVPass final : public llvm::FunctionPass {
     if (parallelMgr) {
       llvm_util_init.reset();
       smt_init.reset();
+      output_mutex.lock();
```

Comment:
Some mutex implementations will complain about destroying a locked mutex.
Leaking stuff all over is a pain for debugging, but since we usually don't have such problems, maybe just use _Exit() here as well?

Reply:
sure. the reason I went with thread+mutex here is that I was worried that a unix signal would interact with Z3 or LLVM poorly in some way, a thread seemed safer.
in the signal version of this code, at this point we would simply mask off signals, avoiding leaking the thread+mutex. that may be a better way to go, I have no idea.

Reply:
I had crashes in Z3 in the past when trying to destroy a locked mutex (don't remember which platform, though).
So as-is will crash for sure. FWIW Z3 ATM only catches SIGINT. I've no idea about LLVM.
Not leaking when there's a timeout is just too hard. But at least it would be nice to not leak everything in the non-timeout case (for debugging). Hence calling _exit() in the non-timeout case isn't very nice.
We have at least these 2 alternatives:
 - add some sync mechanism such that the mutex can be unlocked before exit so it doesn't crash when deleted. This is going to look close to what we have in Z3 I guess.
 - Try out signals. Not portable to windows, but I don't think we care.

Reply:
sure, I'll try these out and see which one feels nicer

Reply:
signals are very clean and also I've set it up so that we'll get crashes if anyone is already using SIGALRM